public/app/features/plugins/state/reducers.test.ts

```diff
@@ -14,7 +15,7 @@ describe('pluginsReducer', () => {
   describe('when pluginsLoaded is dispatched', () => {
     it('then state should be correct', () => {
       reducerTester<PluginsState>()
-        .givenReducer(pluginsReducer, { ...initialState })
+        .givenReducer(pluginsReducer as Reducer<PluginsState, AnyAction>, { ...initialState })
```

Comment:
Why all these `as Reducer<PluginsState, AnyAction>` declarations, something that can be improved in `reducerTester`?

Reply:
Very fair question @hugohaggmark, I think it at least deserves an explanatory comment.

I think the `reducerTester` is actually fine, why we needed it was because now we are exporting the `pluginsReducer` conditionally based on a feature flag (if the feature flag is enabled we use the "new" reducer, which has two extra fields, while by default we are loading the "old" one).

Due to this logic TS sees it as a union type which is not fulfilled by the `pluginsReducer` that is used in these "old" tests.

**We manually set the types here because this test is only testing the functionality of the "old" reducer, and it cannot really happen that the old reducer meets "new" reducer state containing the extra fields.**

Sorry for the lengthy explanation, please let me know if something is not clear.

It can easily be though that we didn't take the best approach!

Reply:
Great explanation! I'm always interested in if the `reducerTester` needs to be extended more. Thank you!